### PR TITLE
fix: allow a due date for open waitlist

### PR DESCRIPTION
**Pull request by KrissDrawing**
This PR addresses #4231 

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

Adds additional condition to display `Is there an application due date?` question when open waitlist selected.
Before as a workaround we could just switch to `available units` set due date, and switch to `open waitlist`. Now it can be added from open waitlist

## How Can This Be Tested/Reviewed?

On partners switch `What is the listing availability?` to open waitlist. Now in `Rankings & Results` we should see `Is there an application due date?` question

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [ ] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

**Review by mcgarrye — APPROVED** [review]
LGTM

**Comment by emilyjablonski** [comment]
The ticket is titled `Allow a due date for all listing types` so I'm curious if there should be no if check to show the field at all?

**Comment by mcgarrye** [comment]
Was the newest change verified with product?

**Comment by KrissDrawing** [comment]
@mcgarrye Yes i asked Sarah, it should be visible all the time

File: sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx [hunk 1]
```diff
@@ -273,6 +249,28 @@ const RankingsAndResults = ({ listing, isAdmin }: RankingsAndResultsProps) => {
             </Grid.Row>
           </>
         )}
+        <Grid.Row columns={2} className={"flex items-center"}>
```
Comment (mcgarrye) [inline sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:252]: I think this logic should be moved to `sites/partners/src/components/listings/PaperListingForm/sections/ApplicationDates.tsx`. This question gets hidden when Lottery is selected and is no longer tied to FCFS so should be with the actual date and time fields at the bottom

Comment (emilyjablonski) [reply sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:252]: @mcgarrye I think that would be a design change and Em has been checking out the UX of the toggled on fields already, I think this is a great point that would be part of separate work

Comment (mcgarrye) [reply sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:252]: Very fair. Still should appear when lottery is selected though if we aren't moving it

Comment (emilyjablonski) [reply sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:252]: Very sorry for the back and forth on the AC here @KrissDrawing! After chatting w Sarah and Em, the desired behavior is to remove the `Is there an application due date?` question and always have the due date field be visible and enabled.

Comment (KrissDrawing) [reply sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx]: Not a problem, if checks pass, it should be ready to review

**Review by emilyjablonski — APPROVED** [review]
thank you!

**Pull request merged**